libass/ass_render.h

```diff
@@ -107,48 +107,28 @@ typedef enum {
     EF_KARAOKE_KO
 } Effect;
 
+typedef struct
+{
+    int x_min, y_min, x_max, y_max;
+} Rectangle;
+
 // describes a combined bitmap
 typedef struct {
-    Bitmap *bm;                 // glyphs bitmap
-    unsigned w;
-    unsigned h;
-    Bitmap *bm_o;               // outline bitmap
-    unsigned o_w;
-    unsigned o_h;
-    Bitmap *bm_s;               // shadow bitmap
-    FT_Vector pos;
+    FilterDesc filter;
     uint32_t c[4];              // colors
-    FT_Vector advance;          // 26.6
     Effect effect_type;
     int effect_timing;          // time duration of current karaoke word
     // after process_karaoke_effects: distance in pixels from the glyph origin.
     // part of the glyph to the left of it is displayed in a different color.
-    int be;                     // blur edges
-    double blur;                // gaussian blur
-    double shadow_x;
-    double shadow_y;
-    double frx, fry, frz;       // rotation
-    double fax, fay;            // text shearing
-    double scale_x, scale_y;
-    int border_style;
-    int has_border;
-    double border_x, border_y;
-    double hspacing;
-    unsigned italic;
-    unsigned bold;
-    int flags;
-    int shift_x, shift_y;
+    int first_pos_x;
```

Comment:
It would be nice to separate this line from the preceding comment to reduce confusion, as the comment refers to `effect_timing`.